### ISSUE TITLE
Current time uses should use UTC not localtime

### ISSUE THREAD

**Issue by grobie** [prometheus/promdash#342]
As a user,
I want to get send a link with the current range and until parameters set.

Current option is to click on the current date in the date picker to fill in the date + time. That time represents the user's localtime though, for example for a user in EST(UTC-05:00) setting the dashboard 5 hours into the past.

Expected:
Convert localtime to UTC.

**Comment by stuartnelson3** [comment]
this sounds reasonable to me, but this has been discussed with @juliusv before, and I believe the prometheus web client functions the same way (using localtime).

is this true, j? given that links are shared across time zones, standardizing seems to be prudent

**Comment by juliusv** [comment]
I think the problem is that if you click on a date in the datetimepicker, it chooses the current local time, but PromDash interprets it as UTC. I.e. if I click on a date in Berlin, it sets it to 13:00 (if it's 13:00 in Berlin), but since PromDash is using UTC, that actually means 14:00 Berlin time (so PromDash will try to show data 1h in the future). What I would expect is that it's all happening in UTC, so it'd choose the current time in UTC instead of local time.

**Comment by stuartnelson3** [comment]
-.- not sure what's going on.

the datetimepicker we're using is attempting what it thinks is the UTC time, but is incorrect.
![screen shot 2015-02-18 at 12 32 47](https://cloud.githubusercontent.com/assets/1398104/6246522/4cfe95ee-b76a-11e4-807d-dcbbe518f26d.png)

`date` is the wrong time (an hour in the future), which the datetime picker is then moving back an hour to supposed UTC time.

![image](https://cloud.githubusercontent.com/assets/1398104/6246537/74863680-b76a-11e4-9eb5-8c4d5af1d40d.png)
(this time is an hour ahead of when I took the screencap)

_sigh_

**Comment by grobie** [comment]
Could this be related https://github.com/shutterstock/rickshaw/issues/504 @stuartnelson3?

**Comment by juliusv** [comment]
On Sat, Mar 7, 2015 at 1:09 AM, Tobias Schmidt <notifications@github.com>
wrote:

> Could this be related shutterstock/rickshaw#504
> <https://github.com/shutterstock/rickshaw/issues/504> @stuartnelson3
> <https://github.com/stuartnelson3>?
>
No, the problem happens in the date picker, not in Rickshaw. Clicking on a
date when no time is set yet should set the date picker to today/now in
UTC, but it sets it to today/now in local time (although the result is
actually interpreted as UTC (which it should)).

**Comment by grobie** [comment]
@juliusv Yeah, figured in the meantime and deleted it. The datepicker plugin is ugly to say the best. I don't know why they do all that wrapping of dates.

**Comment by grobie** [comment]
Bump. This is such an annoying bug, can we use a different datepicker maybe? Or fix it? Or check whether the bug was fixed in the datepicker plugin?

**Comment by juliusv** [comment]
Perhaps, yes. For reference, this is the current one we're using:

https://github.com/tarruda/bootstrap-datetimepicker

(it has recent updates)

...and this is another one we could try out:

https://github.com/Eonasdan/bootstrap-datetimepicker
